bokeh/io.py

```diff
@@ -310,7 +315,12 @@ def _show_with_state(obj, state, browser, new, notebook_handle=False):
     shown = False
 
     if state.notebook:
-        comms_handle = _show_notebook_with_state(obj, state, notebook_handle)
```

Comment:
This needs to be handled more thoroughly. The docstring needs to indicate that `notebook_handle=True` only works with jupyter notebooks and, I think it would be better if trying to use `notebook_handle=True` with zeppelin notebooks raises an explicit error (otherwise we risk bug reports when people expect it to work, but it is silently ignored)